eclass/coreos-kernel.eclass

```diff
@@ -131,9 +131,48 @@ shred_keys() {
 	fi
 }
 
+# Populate /lib/modules/$(uname -r)/{build,source}
+prepare-lib-modules-release-dirs() {
+	# build and source must cleaned up to avoid referencing $ROOT
+	rm "${D}/usr/lib/modules/${version}"/{build,source} || die
+
+	# XXX: For some reason tc-arch-kernel is returning x86_64 on > 2.6.24
+	local kernel_arch=$(tc-arch-kernel)
+	if [ "${kernel_arch}" == "x86_64" ]; then
+		kernel_arch="x86"
+	fi
+
+	# Install a stripped source for out-of-tree module builds (Debian-derived)
+	echo source/Makefile; \
+	find source/arch/${kernel_arch} -follow -maxdepth 1 -name 'Makefile*' -print; \
+	find source/arch/${kernel_arch} -follow \( -name 'module.lds' -o -name 'Kbuild.platforms' -o -name 'Platform' \) -print; \
+	find $(find source/arch/${kernel_arch} -follow \( -name include -o -name scripts \) -follow -type d -print) -print; \
+	find source/include source/scripts -follow -print | cpio -pd \
+		--preserve-modification-time \
+		--owner=root:root \
+		--dereference \
+		"${D}/usr/lib/modules/${version}" || die
```

Comment:
@marineam dropped the subshell, this whole thing is in a function now, did you want the chain of finds moved to a function as well?